hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/payload/ExpressionPayload.scala

```diff
@@ -132,12 +133,10 @@ class ExpressionPayload(@transient record: GenericRecord,
           .serialize(resultingRow)
           .asInstanceOf[GenericRecord]
 
-        if (targetRecord.isEmpty || needUpdatingPersistedRecord(targetRecord.get, resultingAvroRecord, properties)) {
-          resultRecordOpt = HOption.of(resultingAvroRecord)
+        resultRecordOpt = if (targetRecord.isEmpty) {
```

Comment:
Should `ExpressionPayload` directly implements `HoodieRecordPayload` instead of `DefaultHoodieRecordPayload` to avoid confusion?

Reply:
That makes sense, but DefaultHoodieRecordPayload extends OverwriteWithLatestAvroPayload which extends BaseAvroPayload. So this might need more thought